Python/odin/app.py

```diff
@@ -52,7 +52,7 @@ def createapprole(cnx, app_id, role):
 
 def addappuser(cnx, app_id, identity_id):
     cnx.assert_module('app')
-    cnx.execute(ADD_APP_USER, (cnx.reference, app_id, identity_id))
+    cnx.execute(ADD_APP_USER, (cnx.reference, app_id, identity_id, cnx.create_reference()))
```

Comment:
We need a new reference here just so that we can ensure that the app user ID is different to the user ID?